### ISSUE TITLE
[TD-81] Improve logging features and provide informative error messages.

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskserver#81]
_Renato Alves on 2014-11-21T02:00:41Z says:_ 

 Currently taskd uses standardized error messages for a series of error situations. Although clear these messages give little information about the context of where and how they happened. Little is also present in the logs.

In the particular case of invalid payload or payload containing syntax or content errors, the error message could indicate which line of the payload and where "approximately" the parser failed to recognize the format or the contained data.

In short, we should do our best to help other developers find their way around interfacing with taskd.

Motivation for this request: [Mirakel Bug 553|https://mirakel.atlassian.net/browse/MIR-553]

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2014-11-21T02:00:41Z
Modified: 2016-12-19T15:02:19Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2016-12-19T15:02:19Z says:_ 

 There have been a series of logging/feedback improvements in 1.2.0, enough to consider this issue addressed.  Further improvements will be in response to specific requests for improvement only.
